evol/population.py

```diff
@@ -84,14 +84,20 @@ def survive(self, fraction=None, n=None, luck=False) -> 'Population':
         return self
 
     def breed(self, parent_picker, combiner, population_size=None) -> 'Population':
-        """breed(parent_picker=f(Population) -> seq[chromosome],
+        """breed(parent_picker=f(Population) -> seq[individuals],
                                              f(*seq[chromosome]) -> chromosome,
                                                                     population_size = None, ** kwargs) <- TODO: kwargs
         """
         if population_size:
             self.intended_size = population_size
+        # we ensure that we only select the same group before breeding starts
+        size_before_breed = len(self.individuals)
         for _ in range(len(self.individuals), self.intended_size):
-            self.individuals.append(combiner(*parent_picker(self)))
+            parents = parent_picker(self.individuals[:size_before_breed])
+            if not hasattr(parents, '__len__'):
+                parents = [parents]
+            chromosomes = [individual.chromosome for individual in parents]
+            self.individuals.append(Individual(init_function=lambda : combiner(*chromosomes)))
```

Comment:
If we don't append to self.individuals we don't need that indexing hack. We may want to make a `new_individuals = []`, append to that, and then `self.individuals += new_individuals`